src/mobile/apps/user/routes.coffee

```diff
@@ -1,15 +1,16 @@
 CurrentUser = require '../../models/current_user'
 
-module.exports.refresh = (req, res) ->
-  return res.redirect("/") unless req.user
-  req.user.fetch
-    error: res.backboneError
-    success: ->
-      req.login req.user, (error) ->
-        if (error)
-          next error
-        else
-          res.json req.user.attributes
+module.exports.refresh = (req, res, next) ->
+  user = req.user
```

Comment:
This is still an in memory ref no matter where the object is coming from. You'd have to create an entirely new instance of `User` based off of `user.attributes` (or better, `user.toJSON`) and pass that along in order to be completely safe.

Reply:
Agreed.